Signal/src/ViewControllers/ConversationView/ConversationInputToolbar.m

```diff
@@ -1150,6 +609,40 @@ - (BOOL)textView:(UITextView *)textView shouldChangeTextInRange:(NSRange)range r
     return YES;
 }
 
+#pragma mark - Text Input Sizing
+
+- (void)addKVOObservers
+{
+    [self.inputTextView addObserver:self
+                         forKeyPath:NSStringFromSelector(@selector(contentSize))
+                            options:NSKeyValueObservingOptionOld | NSKeyValueObservingOptionNew
+                            context:kConversationInputTextViewObservingContext];
+}
+
+- (void)removeKVOObservers
+{
+    @try {
+        [self.inputTextView removeObserver:self
+                                forKeyPath:NSStringFromSelector(@selector(contentSize))
+                                   context:kConversationInputTextViewObservingContext];
+    } @catch (NSException *__unused exception) {
```

Comment:
What's the likely exception here?

Reply:
KVO throws an exception if remove an observer which isn't currently observing.  So either:

* This view is destroyed before the observer is registered.
* Somehow the observer is de-registered twice.

I'm 99% sure this shouldn't happen now - I've greatly simplified the usage of KVO vs. JSQ - but I decided to hold this over at least for a while until we're sure.  I've added a `owsFail()` and a TODO to remove this.